ocs_ci/ocs/resources/backingstore.py

```
@@ -425,6 +441,32 @@ def backingstore_cleanup():
                 try:
+                    # Remove finalizer from rejected backingstores to allow deletion
+                    # when dependent secrets have already been cleaned up
+                    try:
+                        bs_data = OCP(
+                            kind=constants.BACKINGSTORE,
+                            namespace=config.ENV_DATA["cluster_namespace"],
+                        ).get(resource_name=backingstore.name)
+                        if bs_data.get("status", {}).get("phase") == "Rejected":
+                            log.warning(
+                                f"BackingStore {backingstore.name} is in Rejected state. "
+                                "Removing finalizer to allow cleanup."
```

Ruff (0.15.15)

[warning] Logging statement uses f-string

(G004)